### PR TITLE
Check chain id during tx insertion and proposal validation

### PR DESCRIPTION
**Pull request by eerkaijun**
This prevents the execution client from crashing if someone send a transaction with an invalid nonce. Closes https://github.com/monad-crypto/monad-internal/issues/381

### CODE REVIEW

File: monad-node/src/config/mod.rs [hunk 1]
```diff
@@ -27,6 +27,7 @@ pub type SignatureCollectionType =
 pub struct NodeConfig {
     pub name: Option<String>,
     pub network_name: Option<String>,
+    pub chain_id: u64,
```
Comment (eerkaijun) [inline monad-node/src/config/mod.rs:30]: Should we set this as an optional (with a default value) or required field in the node config file?

Comment (ariqchowdhury) [reply monad-node/src/config/mod.rs:30]: I think make it required. Don't think there is a sensible default value that client should use.

File: monad-eth-block-validator/src/lib.rs [hunk 2]
```diff
@@ -57,6 +60,10 @@ impl<SCT: SignatureCollection, RBCT: ReserveBalanceCacheTrait>
         let mut validated_txns: Vec<EthTransaction> = Vec::with_capacity(eth_txns.len());
 
         for (eth_txn, signer) in eth_txns.into_iter().zip(signers) {
+            if eth_txn.chain_id().unwrap_or_default() != self.chain_id {
```
Comment (ariqchowdhury) [inline monad-eth-block-validator/src/lib.rs]: this one too

Comment (eerkaijun) [reply monad-eth-block-validator/src/lib.rs]: oops missed that

**Pull request merged**